{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'm calling SNPs across all samples, so need to use list of all bamfiles"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to determine how to filter the sites, need to calculate read depth. \n",
    "\n",
    "Therkildsen and Palumbi: excluding sites with a total read depth <300 and >3028 (mean depth +2 standard deviations\n",
    "\n",
    "So need to calculate the mean total read depth (over all samples). \n",
    "\n",
    "Doing some other QC just to see"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -P 15 -b /local/home/sarahgw/WGS/Sequences/filelists/All_Mylu.filelist -ref /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa -out /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.qc -trim 0 -C 50 -baq 1 -minMapQ 20 -doQsDist 1 -doDepth 1 -doCounts 1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\t34722506\t24494187\t18781257\t15505279\t13280089\t11733819\t10599736\t9692502\t9000465\t8435610\t7978920\t7583040\t7272834\t6995445\t6755575\t6560813\t6380016\t6230125\t6086109\t5963864\t5854799\t5753869\t5671724\t5591462\t5518181\t5460487\t5402144\t5353746\t5307400\t5281874\t5247629\t5226749\t5195715\t5178369\t5161880\t5140600\t5138194\t5131957\t5122806\t5123212\t5135911\t5137450\t5145039\t5150701\t5152173\t5168505\t5187396\t5199103\t5221822\t5250111\t5278970\t5297134\t5319020\t5350320\t5391335\t5429049\t5463923\t5503061\t5541364\t5584375\t5636101\t5678599\t5719778\t5770230\t5826837\t5878937\t5944000\t5992944\t6054056\t6118908\t6174548\t6236423\t6304204\t6376630\t6444535\t6524525\t6594344\t6674012\t6750969\t6843660\t6921796\t7004729\t7092571\t7175919\t7262555\t7355076\t7451900\t7547228\t7638831\t7725831\t7831129\t7928781\t8037637\t8137687\t8235763\t8347101\t8454697\t8556569\t8670231\t1168380830\t\r\n"
     ]
    }
   ],
   "source": [
    "!head /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.qc.depthGlobal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It binned anything over 100. so, that's good half of the sites have coverage over 100, but we don't know the upper limit. I'm just going to set upper limit to 10,360 since that would mean that each individual would have greater than 10x depth \n",
    "Not setting lower limit since I'm setting -nInd to half the total number, so should eliminate anything that has super low depth"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Redo SNP calling removing Myse samples:\n",
    "\n",
    "'NJ_U_04_pairedplus_marked_dupes.bam' \n",
    "\n",
    "'NJ_U_14_pairedplus_marked_dupes.bam'\n",
    "\n",
    "'VT_U_05_pairedplus_marked_dupes.bam' \n",
    "\n",
    "'VT_U_03_pairedplus_marked_dupes.bam' \n",
    "\n",
    "call SNPs with filters across all samples. only use sites that have information for at least half the individuals, I think this will be more helpful  \n",
    "\n",
    "\n",
    "add -dogeno 32 and -dopost to this command so you calculate posterior genotype probabilities at the same time.\n",
    "\n",
    "\n",
    "Trying multiple methods for allele frequency estimation just to compare these to the individual population allele freqs. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/All_Mylu.filelist -dogeno 32 -dopost 1 -GL 1 -doMajorMinor 1 -doMaf 11 -doCounts 1 -minQ 20 -minMapQ 30 -P 20 -SNP_pval 1e-6 -minInd 68 -setMaxDepth 10360 -out /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8255384 /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs.gz\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now calculate minor allele frequencies for each population for SNPS, first make sites index file\n",
    "\n",
    "Include major and minor, so that these are consistent across populations, even if not accurate in a given (sub)population"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!cut -f1-4 /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs > /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu.mafs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So, we now have 92,371,964 SNPs!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!tail -n+2 /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites.txt > /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd sites index All_mylu_sites1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for POP in NJ_U NY_U VT_U; do echo $POP; ~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/\"$POP\".filelist -GL 1 -doMajorMinor 3 -doMaf 11 -doCounts 1 -minQ 20 -minMapQ 30 -P 30 -setMaxDepth 10360 -sites /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt -out /local/home/sarahgw/WGS/SNPS/All_mylu/\"$POP\"; echo \"mafs done amphiprion\" | mailx gw.sarah@gmail.com; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for POP in NJ_E NY_E VT_E ; do echo $POP; ~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/\"$POP\".filelist -GL 1 -doMajorMinor 3 -doMaf 11 -doCounts 1 -minQ 20 -minMapQ 30 -P 30 -setMaxDepth 10360 -sites /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt -out /local/home/sarahgw/WGS/SNPS/All_mylu/\"$POP\"; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Redo the NJ and VT unexposed without myse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for POP in NJ_U VT_U; do echo $POP; ~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/\"$POP\".filelist -GL 1 -doMajorMinor 3 -doMaf 11 -doCounts 1 -minQ 20 -minMapQ 30 -P 20 -setMaxDepth 10360 -sites /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt -out /local/home/sarahgw/WGS/SNPS/All_mylu/\"$POP\"; echo \"mafs done amphiprion\" | mailx gw.sarah@gmail.com; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/VT_U.filelist -GL 1 -doMajorMinor 3 -doMaf 11 -doCounts 1 -minQ 20 -minMapQ 30 -P 20 -setMaxDepth 10360 -sites /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt -out /local/home/sarahgw/WGS/SNPS/All_mylu/VT_U.mafs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "You need to:\n",
    "calculate changes in allele frequency--SNP with highest change??\n",
    "summarize differences in allele frequenceis between pops for mitochondrial and nuclear regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_E.mafs.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_U.mafs.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromo\tposition\tmajor\tminor\tknownEM\tunknownEM\tphat\tnInd\r\n",
      "GL429767\t54\tG\tT\t0.000000\t0.000004\t0.000000\t15\r\n",
      "GL429767\t78\tG\tA\t0.109899\t0.109719\t0.119721\t13\r\n",
      "GL429767\t83\tG\tT\t0.000000\t0.000004\t0.000000\t14\r\n",
      "GL429767\t104\tA\tT\t0.207668\t0.207671\t0.232022\t11\r\n",
      "GL429767\t109\tT\tC\t0.056747\t0.055230\t0.024690\t12\r\n",
      "GL429767\t112\tG\tA\t0.000000\t0.000002\t0.000000\t12\r\n",
      "GL429767\t157\tT\tC\t0.000000\t0.000002\t0.000000\t13\r\n",
      "GL429767\t184\tC\tG\t0.000000\t0.000002\t0.000000\t16\r\n",
      "GL429767\t189\tG\tT\t0.000000\t0.000003\t0.000000\t17\r\n"
     ]
    }
   ],
   "source": [
    "!head /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_E.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_E.mafs\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_E.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92371953 /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_U.mafs\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_U.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!paste /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_E.mafs /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_U.mafs | awk '{print $1\",\"$2\",\"$3\",\"$4\",\"$5\",\"$6\",\"$8\",\"$13\",\"$14\",\"$16}'>/local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NY_E.mafs.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NY_U.mafs.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/NY_E.mafs\n",
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/NY_U.mafs\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/NY_E.mafs\n",
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/NY_U.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!paste /local/home/sarahgw/WGS/SNPS/All_mylu/NY_E.mafs /local/home/sarahgw/WGS/SNPS/All_mylu/NY_U.mafs | awk '{print $1\",\"$2\",\"$3\",\"$4\",\"$5\",\"$6\",\"$8\",\"$13\",\"$14\",\"$16}'>/local/home/sarahgw/WGS/SNPS/All_mylu/NY_EU.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/VT_E.mafs.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/VT_U.mafs.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/VT_E.mafs\n",
      "92371964 /local/home/sarahgw/WGS/SNPS/All_mylu/VT_U.mafs\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/VT_E.mafs\n",
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/VT_U.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!paste /local/home/sarahgw/WGS/SNPS/All_mylu/VT_E.mafs /local/home/sarahgw/WGS/SNPS/All_mylu/VT_U.mafs | awk '{print $1\",\"$2\",\"$3\",\"$4\",\"$5\",\"$6\",\"$8\",\"$13\",\"$14\",\"$16}'>/local/home/sarahgw/WGS/SNPS/All_mylu/VT_EU.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU.mafs', 'r')\n",
    "o=open('/local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU2.mafs', 'w')\n",
    "for line in f:\n",
    "    scount=line.count(',')\n",
    "    if scount==9:\n",
    "        o.write(line)\n",
    "    elif 0<=scount<9:\n",
    "        groups=line.split(',')\n",
    "        lineplus=','.join(groups[:2])+''.join([',']*(9-scount))+','+ ','.join(groups[2:])\n",
    "        o.write(lineplus)  \n",
    "o.close() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAPE02072767,3095,T,A,0.470258,0.470257,16,,,\r\n",
      "AAPE02072767,3098,C,T,0.000000,0.000002,16,,,\r\n",
      "AAPE02072780,302,G,A,0.000000,0.000001,16,,,\r\n",
      "AAPE02072780,365,G,C,0.000000,0.000002,14,,,\r\n",
      "AAPE02072780,367,A,C,0.048082,0.048083,14,,,\r\n",
      "AAPE02072780,368,C,T,0.000000,0.000002,14,,,\r\n",
      "AAPE02072780,369,T,C,0.000000,0.000004,14,,,\r\n",
      "AAPE02072780,370,T,A,0.066391,0.066390,14,,,\r\n",
      "AAPE02072780,372,T,C,0.000000,0.000003,14,,,\r\n",
      "AAPE02072780,374,C,A,0.000000,0.000001,14,,,\r\n"
     ]
    }
   ],
   "source": [
    "!tail /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU2.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "109424282 /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU.mafs\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_EU.mafs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In R notebook Angsd_all_SNP_graphs, find SNPs that have grater than 50% change in allele frequency  in all 3 populations. use grep to get 1kb on either side surrounding regions, blast. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429793' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 3618488-3620488 >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429793_3619488"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429805' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 5254331-5256331   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429805_5255331  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429830' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 4775068-4777068   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429830_4776068  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429833' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 4826830-4828830   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429833_4827830"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429835' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 4131453-4133453   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429835_4132453"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429848' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 4053536-4055536   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429848_4054536"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429859' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 3283246-3285246   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429859_3284246"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL429941' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 1538533-1540533   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL429941_1539533"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL430044' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 239397-241397   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL430044_240397"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "grep 'GL430199' -A 10000000 /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa | tr -d \"\\n\\r\" | cut -c 597814-599814   >/local/home/sarahgw/WGS/SNPS/All_mylu/snps/GL430199_598814"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GL429793\t3619488\n",
    "GL429805\t5255331\n",
    "GL429830\t4776068\n",
    "GL429833\t4827830\n",
    "GL429835\t4132453\n",
    "GL429848\t4054536\n",
    "GL429859\t3284246\n",
    "GL429941\t1539533\n",
    "GL430044\t240397\n",
    "GL430199\t598814"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create VCF file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -b /local/home/sarahgw/WGS/Sequences/filelists/All_Mylu.filelist -dovcf 1 -gl 1 -dopost 1 -domajorminor 3 -domaf 1 -sites /local/home/sarahgw/WGS/SNPS/All_mylu/All_mylu_sites1.txt -out /local/home/sarahgw/WGS/SNPS/All_mylu_postv "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make pca from geno prob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/All_mylu.geno.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/ngsTools/ngsPopGen/ngsCovar -probfile /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/All_mylu.geno.gz -outfile /local/home/sarahgw/WGS/SNPS/All_mylu_post.geno -nind 132 -nsites 92371964 -call 0 -norm 0 &> /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NJ_E_01_pairedplus_marked_dupes.bam\n",
      "NJ_E_02_pairedplus_marked_dupes.bam\n",
      "NJ_E_04_pairedplus_marked_dupes.bam\n",
      "NJ_E_23_pairedplus_marked_dupes.bam\n",
      "NJ_E_24_pairedplus_marked_dupes.bam\n",
      "NJ_E_25_pairedplus_marked_dupes.bam\n",
      "NJ_E_26_pairedplus_marked_dupes.bam\n",
      "NJ_E_28_pairedplus_marked_dupes.bam\n",
      "NJ_E_29_pairedplus_marked_dupes.bam\n",
      "NJ_E_30_pairedplus_marked_dupes.bam\n",
      "NJ_E_05_pairedplus_marked_dupes.bam\n",
      "NJ_E_06_pairedplus_marked_dupes.bam\n",
      "NJ_E_07_pairedplus_marked_dupes.bam\n",
      "NJ_E_08_pairedplus_marked_dupes.bam\n",
      "NJ_E_10_pairedplus_marked_dupes.bam\n",
      "NJ_E_22_pairedplus_marked_dupes.bam\n",
      "NJ_E_13_pairedplus_marked_dupes.bam\n",
      "NJ_E_14_pairedplus_marked_dupes.bam\n",
      "NJ_E_15_pairedplus_marked_dupes.bam\n",
      "NJ_E_16_pairedplus_marked_dupes.bam\n",
      "NJ_E_18_pairedplus_marked_dupes.bam\n",
      "NJ_E_21_pairedplus_marked_dupes.bam\n",
      "NJ_U_01_pairedplus_marked_dupes.bam\n",
      "NJ_U_02_pairedplus_marked_dupes.bam\n",
      "NJ_U_03_pairedplus_marked_dupes.bam\n",
      "NJ_U_05_pairedplus_marked_dupes.bam\n",
      "NJ_U_06_pairedplus_marked_dupes.bam\n",
      "NJ_U_08_pairedplus_marked_dupes.bam\n",
      "NJ_U_09_pairedplus_marked_dupes.bam\n",
      "NJ_U_10_pairedplus_marked_dupes.bam\n",
      "NJ_U_11_pairedplus_marked_dupes.bam\n",
      "NJ_U_12_pairedplus_marked_dupes.bam\n",
      "NJ_U_15_pairedplus_marked_dupes.bam\n",
      "NJ_U_16_pairedplus_marked_dupes.bam\n",
      "NJ_U_21_pairedplus_marked_dupes.bam\n",
      "NJ_U_22_pairedplus_marked_dupes.bam\n",
      "NJ_U_23_pairedplus_marked_dupes.bam\n",
      "NJ_U_25_pairedplus_marked_dupes.bam\n",
      "NJ_U_07_pairedplus_marked_dupes.bam\n",
      "NJ_U_17_pairedplus_marked_dupes.bam\n",
      "NJ_U_18_pairedplus_marked_dupes.bam\n",
      "NJ_U_19_pairedplus_marked_dupes.bam\n",
      "NY_E_15_pairedplus_marked_dupes.bam\n",
      "NY_E_16_pairedplus_marked_dupes.bam\n",
      "NY_E_17_pairedplus_marked_dupes.bam\n",
      "NY_E_18_pairedplus_marked_dupes.bam\n",
      "NY_E_27_pairedplus_marked_dupes.bam\n",
      "NY_E_28_pairedplus_marked_dupes.bam\n",
      "NY_E_29_pairedplus_marked_dupes.bam\n",
      "NY_E_30_pairedplus_marked_dupes.bam\n",
      "NY_E_19_pairedplus_marked_dupes.bam\n",
      "NY_E_21_pairedplus_marked_dupes.bam\n",
      "NY_E_22_pairedplus_marked_dupes.bam\n",
      "NY_E_23_pairedplus_marked_dupes.bam\n",
      "NY_E_25_pairedplus_marked_dupes.bam\n",
      "NY_E_03_pairedplus_marked_dupes.bam\n",
      "NY_E_04_pairedplus_marked_dupes.bam\n",
      "NY_E_05_pairedplus_marked_dupes.bam\n",
      "NY_E_06_pairedplus_marked_dupes.bam\n",
      "NY_E_07_pairedplus_marked_dupes.bam\n",
      "NY_E_08_pairedplus_marked_dupes.bam\n",
      "NY_E_09_pairedplus_marked_dupes.bam\n",
      "NY_E_10_pairedplus_marked_dupes.bam\n",
      "NY_U_10_pairedplus_marked_dupes.bam\n",
      "NY_U_11_pairedplus_marked_dupes.bam\n",
      "NY_U_12_pairedplus_marked_dupes.bam\n",
      "NY_U_14_pairedplus_marked_dupes.bam\n",
      "NY_U_15_pairedplus_marked_dupes.bam\n",
      "NY_U_16_pairedplus_marked_dupes.bam\n",
      "NY_U_18_pairedplus_marked_dupes.bam\n",
      "NY_U_19_pairedplus_marked_dupes.bam\n",
      "NY_U_21_pairedplus_marked_dupes.bam\n",
      "NY_U_23_pairedplus_marked_dupes.bam\n",
      "NY_U_25_pairedplus_marked_dupes.bam\n",
      "NY_U_26_pairedplus_marked_dupes.bam\n",
      "NY_U_27_pairedplus_marked_dupes.bam\n",
      "NY_U_28_pairedplus_marked_dupes.bam\n",
      "NY_U_30_pairedplus_marked_dupes.bam\n",
      "NY_U_01_pairedplus_marked_dupes.bam\n",
      "NY_U_02_pairedplus_marked_dupes.bam\n",
      "NY_U_03_pairedplus_marked_dupes.bam\n",
      "NY_U_04_pairedplus_marked_dupes.bam\n",
      "NY_U_05_pairedplus_marked_dupes.bam\n",
      "NY_U_06_pairedplus_marked_dupes.bam\n",
      "NY_U_07_pairedplus_marked_dupes.bam\n",
      "VT_E_10_pairedplus_marked_dupes.bam\n",
      "VT_E_11_pairedplus_marked_dupes.bam\n",
      "VT_E_12_pairedplus_marked_dupes.bam\n",
      "VT_E_13_pairedplus_marked_dupes.bam\n",
      "VT_E_14_pairedplus_marked_dupes.bam\n",
      "VT_E_15_pairedplus_marked_dupes.bam\n",
      "VT_E_16_pairedplus_marked_dupes.bam\n",
      "VT_E_17_pairedplus_marked_dupes.bam\n",
      "VT_E_26_pairedplus_marked_dupes.bam\n",
      "VT_E_28_pairedplus_marked_dupes.bam\n",
      "VT_E_29_pairedplus_marked_dupes.bam\n",
      "VT_E_30_pairedplus_marked_dupes.bam\n",
      "VT_E_01_pairedplus_marked_dupes.bam\n",
      "VT_E_25_pairedplus_marked_dupes.bam\n",
      "VT_E_18_pairedplus_marked_dupes.bam\n",
      "VT_E_19_pairedplus_marked_dupes.bam\n",
      "VT_E_21_pairedplus_marked_dupes.bam\n",
      "VT_E_22_pairedplus_marked_dupes.bam\n",
      "VT_E_24_pairedplus_marked_dupes.bam\n",
      "VT_E_02_pairedplus_marked_dupes.bam\n",
      "VT_E_03_pairedplus_marked_dupes.bam\n",
      "VT_E_04_pairedplus_marked_dupes.bam\n",
      "VT_E_05_pairedplus_marked_dupes.bam\n",
      "VT_E_06_pairedplus_marked_dupes.bam\n",
      "VT_E_07_pairedplus_marked_dupes.bam\n",
      "VT_E_08_pairedplus_marked_dupes.bam\n",
      "VT_E_09_pairedplus_marked_dupes.bam\n",
      "VT_U_09_pairedplus_marked_dupes.bam\n",
      "VT_U_10_pairedplus_marked_dupes.bam\n",
      "VT_U_11_pairedplus_marked_dupes.bam\n",
      "VT_U_01_pairedplus_marked_dupes.bam\n",
      "VT_U_02_pairedplus_marked_dupes.bam\n",
      "VT_U_27_pairedplus_marked_dupes.bam\n",
      "VT_U_29_pairedplus_marked_dupes.bam\n",
      "VT_U_31_pairedplus_marked_dupes.bam\n",
      "VT_U_20_pairedplus_marked_dupes.bam\n",
      "VT_U_22_pairedplus_marked_dupes.bam\n",
      "VT_U_25_pairedplus_marked_dupes.bam\n",
      "VT_U_26_pairedplus_marked_dupes.bam\n",
      "VT_U_12_pairedplus_marked_dupes.bam\n",
      "VT_U_14_pairedplus_marked_dupes.bam\n",
      "VT_U_15_pairedplus_marked_dupes.bam\n",
      "VT_U_17_pairedplus_marked_dupes.bam\n",
      "VT_U_19_pairedplus_marked_dupes.bam\n",
      "VT_U_33_pairedplus_marked_dupes.bam\n",
      "VT_U_37_pairedplus_marked_dupes.bam\n",
      "VT_U_40_pairedplus_marked_dupes.bam\n"
     ]
    }
   ],
   "source": [
    "!head -n 132 /local/home/sarahgw/WGS/Sequences/filelists/All_Mylu.filelist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!Rscript -e 'write.table(cbind(seq(1,132),rep(1,132),c(rep(\"NJ_E\",22),rep(\"NJ_U\",20),rep(\"NY_E\",21), rep(\"NY_U\",22), rep(\"VT_E\",27), rep(\"VT_U\",20))), row.names=F, sep=\"\\t\", col.names=c(\"FID\",\"IID\",\"CLUSTER\"), file=\"/local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/ALL_mylu.clst\", quote=F)'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "123\t1\tVT_U\r\n",
      "124\t1\tVT_U\r\n",
      "125\t1\tVT_U\r\n",
      "126\t1\tVT_U\r\n",
      "127\t1\tVT_U\r\n",
      "128\t1\tVT_U\r\n",
      "129\t1\tVT_U\r\n",
      "130\t1\tVT_U\r\n",
      "131\t1\tVT_U\r\n",
      "132\t1\tVT_U\r\n"
     ]
    }
   ],
   "source": [
    "!tail /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/ALL_mylu.clst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.9 1.64 1.56 1.49 1.41 1.35 1.3 1.29 1.24 1.23 1.18 1.16 1.14 1.13 1.12 1.11 1.1 1.08 1.07 1.06 1.05 1.04 1.04 1.02 1.01 1.01 1 0.992 0.985 0.968 0.965 0.958 0.957 0.955 0.951 0.948 0.938 0.929 0.927 0.921 0.917 0.911 0.901 0.892 0.887 0.883 0.878 0.872 0.867 0.864 0.861 0.86 0.858 0.844 0.838 0.817 0.795 0.787 0.783 0.78 0.772 0.766 0.761 0.753 0.751 0.74 0.73 0.724 0.714 0.702 0.697 0.695 0.687 0.678 0.669 0.655 0.647 0.644 0.637 0.631 0.624 0.618 0.618 0.616 0.611 0.605 0.6 0.594 0.589 0.582 0.575 0.57 0.564 0.559 0.553 0.544 0.534 0.532 0.528 0.524 0.521 0.517 0.513 0.504 0.499 0.494 0.488 0.476 0.466 0.457 0.45 0.446 0.443 0.422 0.41 0.398 0.384 0.372 0.364 0.354 0.339 0.329 0.324 0.319 0.31 0.293 0.284 0.27 0.248 0.246 0.244 0.000714 \n",
      "Saving 7 x 7 in image\n"
     ]
    }
   ],
   "source": [
    "!Rscript /local/home/sarahgw/ngsTools/Scripts/plotPCA.R -i /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/All_mylu.covar -c 1-2 -a /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/ALL_mylu.clst -o /local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/ALL_mylu.pca.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAfcAAAH3CAMAAACRq+YVAAACkVBMVEX////MzMwAAABVVVXd3d1m\nZmYRERHu7u6IiIi7u7siIiJ3d3dERESZmZmqqqozMzP+/v76+vr7+/v9/f3s7Ozy8vL39/f19fX8\n/Pz4+PhgYGAODg6Tk5Otra0EBAQKCgrNzc1nZ2cGBgYICAg0NDQ5OTns593otkXnowznpxnr3b3p\nvFjmnwDnrSrnowrqyoXnoATosTSt2Mw0sI7N4twGoHYAnnMOo3vd5+RgvqUKonkKonh2xa9swquO\nzby93dS9vb0ePjUBiWQCmG8HoHdYvKAAgmwAfp4AiYsAb1BBYVg+lIIAcrIAfKMIJx+11sknRyQM\neayIdUsNeawHdqzc3NxxcXH09PR8fHzQ0NChw2uFvbOgoKAPeasTfKdpaWkOeavu9vqHr2zO3s6H\noV4php9Dk5LExMTo6Oji6u5lqdAmhbxtq85AlMMEdLMHdrQhg7omh71YoMjC2eYoiL40jb/d5Om4\nuLisrKxZWVn5+fnS0tJNTU0mJiampqYPDw/CwsKysrIQEBBaWlri4uIMDAyhoaGXl5cHBwfu0ZHs\nwWLy7eLprSbnogbnpAzw4sLx59LtyXnrv1rv3LLe6vHJ4vCQzO5dt+tvvuxbtutWtOqk0+/T5vFe\nuOuWzu5yv+xatuvo7vK02u+az+729vaX08MRpHwMo3pavqPi7eoEoHWh1siy3NAQpHsPo3vC4tnS\n5+FNuZwrrYqm2Mvy8dvx6Xrx517y77vx5U3w5ELx5Uby8M/y78Px6oTy8ufx5Uvx5UXy7rjy7rDx\n5U7x5UfS4eqy0OFaossKeLUIdrR5stIMebZipswrib4GdrSRv9lpqs4qKioJCQlYWFiOjo52dnZ+\nfn7JycnX19dlZWWUlJTo+ZTNAAAAAWJLR0QAiAUdSAAAAAlwSFlzAAAASAAAAEgARslrPgAAEHNJ\nREFUeNrtnYl/69hZho8V75atScJk7mWYUloolKZTaGdCKduUfQk7kWUwXPZk2Eop4BmgFCgt0Jah\nQKEwUKCsZd+XC5S9UChroUD5aziSbCdOJMe5kaLlfd5fIts6ymNdP/f7LCXHtjGEEEIIIQlprL2Z\nLZwUEyfMlpXRbLWddqsTrmsvR7s9u+jPb3RajjOw4lzHtdsP7VUzGC223Ook0wZtx+0k0cyo7/Sj\nH2+G99dot8Mfc1egJLc49rsz8OyD3+2Yhmcf94HrLAY7nvW7vNlqdTqtgXEHdhPTiWyGG8RbbiXT\nhv1muHUCrec1TdcJIdEqr9ezw+F/mVMoyS/OfNkKy9c07XIwWHoPa3q4vNkOJXnRTzhmMIzWLWpz\nOEymDQbLtedoW735le4gJtqvXmsFSvLLokLbzdV1UWI3i5tRcTpxvTfj+jaj1nzLTirN9PpJtLZ9\nBgh/qul15vXeN/NnhAWU5JfoGdk+Xztn1y2u9Dtnb0Z93omf31vzau3ETudtPpE2bI+SaE67ZwZW\nsNuLVjX6/UZ0yHAKJTlmWX5xhQ6aZ9Yt23YcK709b9O2IQ/b7eFycN7mE2gN1+sl0sJDRFvp3a3l\nKvu8vgIlOWZZzFGtjZw13sM04mO0raZpNxrt5eC8zV+k2QM8k0zzbK132vagLmoR4Rr7tL4CJTlm\n8RBHR+A9b3B2nfGay5udnhl6nY4bFXbYkGNFzejYe9HmL9IGg9O7WqUNbJ9vtc78VHgOdxZK8szS\ncXPLcfqD1XWt0fJmzy4Hts+H16NzuLglx0/JizZ/kRaXcwrNaXXO/FR4DncWSgrL6FIB8fFds3M5\naxPaCpQUl8sabtPNknZPUJJDepdUXvdKv22/jHZPUEIIIYQQQmRz3/Zqdna2N80VtgSant1CvG+/\nz2ruP78iPXsbbwl0TR7AuyQU75pQvGtC8a4JxbsmFO+aULxrQvGuCS3G+87earZ39zbNrY23BLom\nt6l3SSh9XhOKd00o3jWheNeE4l0TindNKN41oXgXgL7vgw++37lRvNcf+tAzHOf9n7k6ivf6Qz/A\ncZxnPbQ6ivf6Q5/5bMd5sLk6incB6EMf+EHn2jzeRaF414TiXROKd00o3jWheNeE4l0TindNKN41\noXjXhOK93tDnfPCHPPdDE0bxXm/o8/b395+bMIr3ekOfb70/nDCK93pDX/Bh+/sfnjCK95pDX/D8\n5z0nYRTvmlC8a0LxrgnFuyYU7/WGvvBFL3okaRTvtYY+8ujBwUe8OGEU77WGfuTBwcGjL0wYxXud\noS9+ifX+UY8kjOK9ztCPttoPPiZpFO91hn6sfXp/9OOSRvFeY+hjL/34T/jET0ocxXt9oc1P/pRP\n/bRPTx7Fe32hn3F4ePiZn5U8ivfaQj/7MMznJI/ivbbQz428f17yKN7rCv38SPsXpIzm7v3I98f2\nYhz4k3iB9xuBfmGk/fCLUoZz9x6MJ1N7MZ2Mg3iB9xuBfnGk/UvShvP3Hn3FF/Prnncn73slXxp5\n/7K04Z287z/BO/V+A9AvD7V/Reowfb6m0OZXftVXrxnO3fux75/YKj/x/eN4gfcyQDmP04SWw/vj\nX/O1X1eHR7M60HJ4//rDw2/YUHypH83qQMvh/WX22PMba/BoVgdaDu8vt96/qQaPZnWg5fD+im9+\n2bd8aw0ezepAy+G9Lo9mdaB414TiXROKd00o3jWheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF4\n14TiXROKd00o3jWheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeE4l0T\nindNKN41oXjXhOJdE4p3TSjeNaHFeN/ZW8327t6mubXxlkDX5Db1Lgmlz2tC8a4JxbsmFO+aULxr\nQvGuCcW7JhTvmlC8a0LxrgnFuyYU75pQvGtC8a4JxbsmFO+aULxrQvGuCcW7JhTvmlC8a0LxrgnF\nuyYU75pQvGtC8a4JxbsmFO+aULxrQvGuCcW7JhTvmlC8a0LxrgnFuyYU75pQvGtC8a4JxbsmFO+a\nULxrQvGuCcW7JjR370e+P7YXx74/MzM/mOK9FNDcvQfjSeh6Oja+mR5R7yWB5u89+rIZB6Yb2KI3\nxvPu5H2v5JLs5IeeBTZL7090w3Zviz4M9V409Mb6fFjoM7yXBZq7d3tAd2Ir/smw9mdxn8d78VDO\n4zSheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeEJnh3XLc/xHu9oUne\njWn08V5vaLJ3+91wXbdhWtES77WDJnsfemZraIZbxomWeK8dNPn5fasxr3pnXv94rxk0pc8bvNcb\nmuqdPl9raKr3+LjOabl9jutqCL3k9zY59Hi8lwGKd03oJd4HeK8nlN/Pa0Ivev+2c7kP7zWEXvR+\n/7ns4r2GULxrQvGuCcW7JnQj7254OufOv8MLm2ud4eG9aGia92//jld+56n3dm/VO/VedWia91c5\nzrO+a+m96+G9XtAU79/9PY7jvHrpPez05/r8tdzjvWhoivfvfY3jPOO1p9477Qb1XidoWp//vvb3\nv/r0+d2Ykee6HbzXBrrZ8bxd3XKt+iF9viZQzt81oVf03huEwXvlodS7JhTvmtDL//6eg3a8Fw69\n6H33XPLwvrO3mu3dvU1za+Mtga7J7QtO6PMKUJ7fNaF414TiXRN6b/MujLneL+nxXjQ01fsPvO71\n6+dd4L3K0DTvb/jB/adev3beBd6rDE3x/kNP7e/vvy513gXeqw5N8f7Gh633H1477wLvVYam9fkf\nefipN6TOu2g1jOm18F5haOpx3Rt/NH3eRWfLdbc6eK8wlPN3TSjzLjSh1LsmFO+aULxrQrN9n5NG\nvz3c5Mwe70VDs613r9HsN/FeAWi23q3xYQvvFYBmXO8d+93Ee/mh2Xof9Edm1L/8QwvwXjT0CvMu\nRi1jOl7869m267pJn0Uy6oW/28F76aGp3t/0Y2e8x/Mu3J5pjeKN0lp5+AebDd7HGO9FQ9O8//ib\n3/wTp97jeRfNreVTd7L3Rt+u723woUR4Lxqa4v1NP/n000//1Pl5FwNvUcvJfT4e7nh4Lz00xftP\nv8V6/5m0eRdp9b7x2yLgvWhoWp//2Z97+i1vXZ13san3Dsfz5YemHtf9/C+89dy8i7O624nvYNcN\nP3OsscFb2+G9aGi25+8jz3G80eXb4b1o6JW9RxMvelf4v4D3MkIzrfdG39nsU4jwXjQ0U+/9nj17\nx3sVoJl6Xzn0w3uZoRe933cum2vHe3WgF73/4rlcQTzeKwPNtM878+C99FDmVWpC8a4Jxbsm9B7n\n22x8BIf3ckLTvP/SL//K2y6Zb4P3CkNTvP/qSw4Ofu3X18+3wXuFoSnef+M3Dw4O3pY+3wbvFYem\n9fnfOjh49LfXz7fBe4Whad5/53d/7/fPzrs4P9+mYzqbeT/y/bG9mPnB1IwDf4L3UkCv8PkyZ703\nPNfb6C+uJhhPpvZiehQuJuMA76WA5n7+HkRfphv4s8V1z7tzPSa5dnYurMlsvs0ssFm4tvFPr1Pv\nRUPzr/e4z89C7/T50kBz937s+ye2ymdhnz/x/WO8lwJazOeM4L1o6OWfK3Qf3msI5e9xmlC8a0Lx\nrgnFuyZ0I+/hZIvu0HXbi0//TnubE7xXBZrm/Q/+8I/+eOm9O4zdX/IyaLxXB5rm/U/u3v3TP1t4\nb26FX3ivETTF+59b73ffvnx+9zrDrjn38nf6fJWhKd7/4i/v3v2rv1567w63mtR7naBpff5v/vbv\n3n56PN/c8lZ0473q0A3P47zuiu7ktznBe3WgnL9rQq/qPZO3OcF74VDqXROKd03o5X9/z0E73guH\nMt9GE0qf14TiXROKd00o3jWh9zbv4povh8V74dBU7+/4+1PvifMu8F5laJr3f3jnO1++9J447wLv\nVYameH/HPx4eHv5T6rwLvFcdmuL9Fe+y3v957bwLvFcZmtbn/+VfD9/1ePq8i43f7gLv5YSmHtf9\n278/nj7vYvO3u8B7OaGcv2tCmXehCaXeNaF414TiXRPKvAtN6APXl4j3CkLxrgnFuyYU75pQvGtC\n8a4JxbsmtBjvO3ur2d7d2zS3Nt4S6Jrcpt4lofR5TSjeNaF414TiXROKd00o3jWheNeE4l0TindN\nKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF4\n14TiXROKd00o3jWheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeE5u79\nyPfH9iIIgiePZn4wxXspoLl7D8aT2PXJ1EyPqPeSQPP3Hn3ZdMemG/gze83z7uR9r+SS7OSHntne\nHiy9T56IVvrRknovGnpjfd6Wu5nhvSzQ3L0f+/6JrfhxEHWAqM/jvXgo53GaULxrQvGuCcW7JhTv\nmlC8a0LxrgnFuyYU75pQvGtC8a4JxbsmFO+aULxrQvGuCcW7JhTvmlC8a0LxrgnFuyYU75pQvGtC\n8a4JxbsmFO+aULxrQvGuCcW7JhTvmlC8a0LxrgnFuyYU75pQvGtC8a4JxbsmFO+aULxrQvGuCcW7\nJhTvmlC8a0LxrgktxvvO3mq2d/c2za2NtwS6Jrepd0kofV4TindNKN41oXjXhOJdE4p3TSjeNaF4\n14TiXROKd00o3jWheNeE4l0TindNKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeE4l0T\nindNKN41oXjXhOJdE4p3TSjeNaF414TiXROKd00o3jWheNeE4l0TindNKN41oXivP/Sxxy6O4r32\n0P94zbv/88Io3usO/S/Hcd59oeLz9z7Be6HQ91jvz37o/Gju3o/86GIc+JN4gfebhTYfdJxXXhi9\nKe/TyTiIF3i/YWjzPf99cTT/Ph8slkG8MMbz7uR+r2R9dvJDzwKbJO/Ue/HQm6p3+ny5oDfjPTAn\nvn8cL/BeBijn75pQvGtC8a4JxbsmFO+aULxrQvGuCcW7JhTvmlC8a0LxrgnFuyYU75rQYrzvPrCa\n5uiBTXN74y2Brsl2Id7P53/ymIADtPSpzj9cHJpx7v9foJWAEnLdTK6PINXL8XujiyPfP4oXGWTc\nDbpjexkEwZNH0yCYZgmNeFnv6dT3ZyabPZ3vWkRe4EuYSfzCimA8DuJFBplOzCR+BE+mJhPiGWiw\n3N0Mof9njJ/Rns53LSKfPhDlS9IE+wyQMcj+b39vYEspQ2jEy35Praxs9jQ4pQan+BIlfF3FLGvv\nEXT5z508Ea30M4XOSzPTPR1Pp+Ms9tRUwPvpfpqs+/yxOY7amy338SyTR3MJjXlZ72n3yGS1p4s+\nH5KXD0QJE/+fPA5fU3G8eGHFNWMPZ4KxhcYPgJ9Nn19AI17Ge3ryZPiasmz2NNq1ICbHeEIIIYQQ\nQoiN47puv2HM0HO9YbhiUPQekZuIY797fTNyO6bT75nGwCl6j8hNxIkXni150+uZId41EmoeuMZZ\nWUFqn/D5vdXBu1rmmqM+320ZvItkrnnk2eM6r2fwLpKF5uV5HN4JIYQQQgghVc3/Azop4EnFVV4o\nAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE3LTA5LTAxVDEyOjQ3OjA1LTA0OjAwy94K6AAAACV0RVh0\nZGF0ZTptb2RpZnkAMjAxNy0wOS0wMVQxMjo0NzowNS0wNDowMLqDslQAAAAgdEVYdHBkZjpIaVJl\nc0JvdW5kaW5nQm94ADUwM3g1MDMrMCswczYNegAAABR0RVh0cGRmOlZlcnNpb24AUERGLTEuNCAc\nRzp4AAAASnRFWHRzaWduYXR1cmUAY2Y3Y2E4NDgwN2E4MzdjOWViZjQzYmY0NTI1ZjViOWQ4NmU0\nYmRkYmQ0N2JjM2JiNWI0MTZiNTgzZDMzOGY0NtDxmD0AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<wand.image.Image: cf7ca84 'PDF' (503x503)>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from wand.image import Image as WImage\n",
    "img = WImage(filename='/local/home/sarahgw/WGS/SNPS/All_mylu/genoprob/ALL_mylu.pca.pdf')\n",
    "img"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "separate mafs for mitochondrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "zgrep 'AAPE02072785' NJ_E.mafs.gz > NJ_E_AAPE02072785.mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "zgrep 'AAPE02072785' NJ_U.mafs.gz > NJ_U_AAPE02072785.mafs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "look for associations for each group.\n",
    "Paper on method for determining associations: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-231"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sed -e 's/NJ_E_.._pairedplus_marked_dupes.bam/1/g' -e 's/NJ_U_.._pairedplus_marked_dupes.bam/0/g' NJ.filelist > NJ_pheno.ybin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sed -e 's/NY_E_.._pairedplus_marked_dupes.bam/1/g' -e 's/NY_U_.._pairedplus_marked_dupes.bam/0/g' NY.filelist > NY_pheno.ybin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sed -e 's/VT_E_.._pairedplus_marked_dupes.bam/1/g' -e 's/VT_U_.._pairedplus_marked_dupes.bam/0/g' VT.filelist > VT_pheno.ybin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!awk {print} /local/home/sarahgw/WGS/Sequences/filelists/VT_pheno.ybin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!wc -l /local/home/sarahgw/WGS/Sequences/filelists/NJ_pheno.ybin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "/local/home/sarahgw/angsd/angsd -yBin /local/home/sarahgw/WGS/Sequences/filelists/NJ_pheno.ybin -doAsso 1 -GL 1 -out NJ_assoc -doMajorMinor 1 -doMaf 1 -minQ 20 -minMapQ 30 -SNP_pval 1e-6 -bam /local/home/sarahgw/WGS/Sequences/filelists/NJ.filelist -P 15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "/local/home/sarahgw/angsd/angsd -yBin /local/home/sarahgw/WGS/Sequences/filelists/NY_pheno.ybin -doAsso 1 -GL 1 -out NY_assoc -doMajorMinor 1 -doMaf 1 -minQ 20 -minMapQ 30 -SNP_pval 1e-6 -bam /local/home/sarahgw/WGS/Sequences/filelists/NY.filelist -P 15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "/local/home/sarahgw/angsd/angsd -yBin /local/home/sarahgw/WGS/Sequences/filelists/VT_pheno.ybin -doAsso 1 -GL 1 -out VT_assoc -doMajorMinor 1 -doMaf 1 -minQ 20 -minMapQ 30 -SNP_pval 1e-6 -bam /local/home/sarahgw/WGS/Sequences/filelists/VT.filelist -P 15; echo \"VT Assoc done amphiprion\" | mailx gw.sarah@gmail.com"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ok so if this is true:\n",
    "The likelihood ratio statistic. This statistic is chi square distributed with one degree of freedom\n",
    "Then that means p less than 0.05 is greater than 3.84\n",
    "\n",
    "Right, but you need to do a p adjustment to deal with false positives. Ok, so let's take it into R and calculate p value and correct. \n",
    "Wait, maybe this is already accounting for false positives, paper seems to indicate that, no mention of correcting? Oh, so for standard GWAS there is a set significance thresholde of 5 x 10 ^ -8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_assoc.lrt0.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_assoc.mafs.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/NY_assoc.lrt0.gz\n",
    "!gunzip /local/home/sarahgw/WGS/SNPS/All_mylu/VT_assoc.lrt0.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Going to just try calculating sample allele frequency for snps for two populations, then I can maybe just look for highest changes in allele frequency like they did in tasmanian devil paper: https://www.nature.com/articles/ncomms12684#methods\n",
    "Fitst, need to make indexes from list of snps so that you can input them as -sites file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "/local/home/sarahgw/angsd/misc/realSFS print NJ_E.saf.idx NJ_U.saf.idx -r GL429767:6000000-6000100 > NJSaf_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!awk '{print NF; exit}' /local/home/sarahgw/WGS/SNPS/All_mylu/NJSaf_test "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's try doing doMaf-0 to see if that will give us allele frequencies? No, there is no output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, now let's calculate the posterior genotype probabilities In order to do that we first need to do sfs with E and U separately. Since I'm not doing the whole thing, need to take first line make new sfs file, remember folded sfs has half as many categories as unfolded sfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/misc/realSFS -P 20 NJ_E.saf.idx -nsites 1000000000 > NJ_E_sub.ml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/misc/realSFS -P 20 NJ_U.saf.idx -nsites 1000000000 > NJ_U_sub.ml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "./angsd -dopost 3 -b list -gl 1 -domajorminor 1 -domaf 1 -pest angsdput.saf.idx.ml -dogeno 2 -r 1 -out angsdput2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -dopost 3 -fold 1 -anc /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa -P 10 -b /local/home/sarahgw/WGS/Sequences/filelists/NJ_U.filelist \n",
    "-gl 1 -domajorminor 1 -domaf 1 -pest /local/home/sarahgw/WGS/SNPS/All_mylu/NJ_U_sub1.ml -minMapQ 30 -minQ 20 -dogeno 2 -out NJ_U_posterior"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "~/angsd/angsd -P 10 -b /local/home/sarahgw/WGS/Sequences/filelists/NJ_E.filelist -fold 1 -anc /local/home/sarahgw/references/Myotis_Lucifugus2.0/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa -dopost -sfs NJ_U_sub.ml -gl 1 -minMapQ 30 -minQ 20 -doMajorMinor 1 -out NJ_E_posterior"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
